lightning/src/ln/channel.rs

```diff
@@ -5481,17 +5481,21 @@ impl<SP: Deref> Channel<SP> where
 		// If we're past (or at) the AwaitingChannelReady stage on an outbound channel, try to
 		// (re-)broadcast the funding transaction as we may have declined to broadcast it when we
 		// first received the funding_signed.
-		let mut funding_broadcastable =
+		let mut funding_broadcastable = None;
+		if let Some(funding_transaction) = &self.context.funding_transaction {
 			if self.context.is_outbound() &&
 				(matches!(self.context.channel_state, ChannelState::AwaitingChannelReady(flags) if !flags.is_set(AwaitingChannelReadyFlags::WAITING_FOR_BATCH)) ||
 				matches!(self.context.channel_state, ChannelState::ChannelReady(_)))
 			{
-				self.context.funding_transaction.take()
-			} else { None };
-		// That said, if the funding transaction is already confirmed (ie we're active with a
-		// minimum_depth over 0) don't bother re-broadcasting the confirmed funding tx.
-		if matches!(self.context.channel_state, ChannelState::ChannelReady(_)) && self.context.minimum_depth != Some(0) {
-			funding_broadcastable = None;
+				// Broadcast only if not yet confirmed
+				if self.context.get_funding_tx_confirmation_height().is_none() || self.context.get_funding_tx_confirmed_in().is_none() {
+					funding_broadcastable = Some(funding_transaction.clone())
+				} else {
+					log_debug!(logger,
+						"Not rebroadcasting funding tx, already confirmed, height: {} txid: {}",
+						self.context.get_funding_tx_confirmation_height().unwrap_or_default(), funding_transaction.compute_txid());
```

Comment:
Sorry, what I meant earlier was that it's the typical case that we don't need to rebroadcast the funding transaction, so this would result in excessive (not just verbose) logging. This is because `monitor_updating_restored` is called in the `handle_monitor_update_completion!` macro (i.e., whenever a channel monitor is updated).

Reply:
My reasoning was that there can be 2 ways for non-optimal behavior:
- rebroadcasting when not needed -- this is logged at rebroadcast
- not rebroadcasting when needed -- I added this logging for that case, and it's only debug-level log
But I can remove it to avoid excessive logging.